---
title: 'Genomics at superresolution: Mapping Drug targets on single cell resolution in Fibrosis'
author: ''
date: '2019-09-17'
slug: genomics-at-superresolution
categories: []
tags:
  - scRNA-seq
  - Seurat
  - h5 file
  - R
  - R Markdown
  - Clustering
  - PCA
  - tSNE
  - IPF
subtitle: ''
summary: ''
authors: []
lastmod: '2019-09-17T21:23:26+02:00'
featured: no
image:
  caption: ''
  focal_point: ''
  preview_only: no
projects: []
---

Advances in **microfluidic technologies** enabled us to barcode single cells in lipid droplets and to **resolve genomes of individual cells** from a sequencing mixture (e.g, [10X Genomics](https://www.10xgenomics.com/)). By using **Single cell RNA sequencing (scRNA-seq)** we can discover **rare cell populations**  and genes that are specifically acting in those. Potential is high and the list of publications growing daily.  

* If you are a scientist in a biotech exploring novel targets those might be a great source to **gather specific information**.

[Seurat](https://cran.r-project.org/web/packages/Seurat/index.html) package is a great tool for digging into single cell datasets. It will open you access beyond what is in the publications. You can find many tutorials in their [website](https://satijalab.org/seurat/).


Here, I will focus on a recent [paper](https://www.atsjournals.org/doi/pdf/10.1164/rccm.201712-2410OC) which explored transcriptome of lung cells from Pulmonary fibrosis patients by scRNAseq.

* **Pulmonary fibrosis** is a progressive scarring of the lung tissue leading to death within 3-4 years.
* **Current therapies does not increase the survival** 
* Many Biotech companies are developing **novel drugs**

{{% alert note %}}
What types of cells those drugs are acting on? 
{{% /alert %}}

Our workflow will be in three steps

1. Basic Seurat workflow
2. Identify cell types on a tSNE plot
3. Visualize drug targets on single cell plots

I picked up scRNA-seq data from a patient with Polymyositis associated intersitial lung disease. 

---> [Download Single Cell RNA seq data](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM3489196)


### Seurat Workflow

Load the packages that will be used.
```{r, loading_packages, warning=FALSE, message=FALSE}
library(Seurat)
library(dplyr)
library(ggplot2)

```

### Importing the h5 file
Here the data comes as an **h5 file.** But no worries, we can handle that with **Read10X_h5** function in Seurat. Since the file is directly under my working directory I can apply `Read10X_h5("Filename.h5")`
```{r  warning=FALSE}
MyoILD_01 <- Read10X_h5("posts_data/GSM3489196_Myositis-ILD_01_filtered_gene_bc_matrices_h5.h5")
Fib <- CreateSeuratObject(counts = MyoILD_01, min.cells = 3, project= "FightFibrosis", min.features = 200)
Fib
```
  
```{r  include=FALSE}
a <- dim(Fib)
```
Our Data matrix now contains **`r a[1]`** genes and **`r a[2]`** cells

### A Quick look at the raw count data

```{r, warning = FALSE, message=FALSE}
# Lets examine a few genes in the first thirty cells
MyoILD_01[c("SPP1","TGFB1","CCL2"), 1:30]

```

### Standard pre-processing workflow for scRNA-seq data

Pre-processing involves filtration of cells based on Quality control metrics (e.g .mitochondrial contamination, Coverage).  We will proceed by normalization and identification of the highly variable features then at the end we will scale the data. 


#### QC and selecting cells for further analysis

```{r mito, fig.height=7, fig.width=13}
# Let's add Mitochondrial stats to Fib data. [[ operator can add 
# columns to object metadata. 

Fib[["percent.mt"]] <- PercentageFeatureSet(object = Fib, pattern = "^MT-")
```

```{r qc, fig.height=7, fig.width=13}
# Quick look at the Quality control metrics for the first 5 cells
head(x = Fib@meta.data, 5)
```

#### Violin plots to visualize QC metrics

```{r qc2, fig.height=7, fig.width=13}
#Visualize QC metrics as a violin plot
VlnPlot(object = Fib, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"),
        ncol = 3)

# We will use FeatureScatter to visualize feature-feature relationships like 
# RNA counts or percentage of mitochondiral contamination

plot1 <- FeatureScatter(object = Fib, feature1 = "nCount_RNA", feature2 = "percent.mt") 
plot2 <- FeatureScatter(object = Fib, feature1 = "nCount_RNA", feature2 = "nFeature_RNA") 
CombinePlots(plots = list(plot1,plot2))
# As you see on the left plot cells with high percentage of mitochondrial genes
# have very low numbers of RNA indicating that they are low quality/dead cells.
# Let's remove them.
Fib <- subset(x = Fib, subset = nFeature_RNA > 200 & nFeature_RNA < 4000 & 
              percent.mt < 12.5)
```


#### Normalize the data

Log normalization helps to reduce the influences of the outliers. 
Normalized values will be stored in `Fib[["RNA"]]@data`.

```{r normalize, warning = FALSE, message=FALSE}
Fib <- NormalizeData(object = Fib, normalization.method = "LogNormalize", 
                     scale.factor = 1e4)
```

#### Identify highly variable features

Let's calculate the features that exhibit high cell-to-cell variation in the dataset. Focusing on those genes will help to highlight biological signal.
```{r var_features, fig.height=5, fig.width=11, warning = FALSE, message=FALSE}
Fib <- FindVariableFeatures(object = Fib, selection.method = 'vst',  nfeatures = 2000)

# Identify the 10 most highly variable genes
top10 <- head(x = VariableFeatures(object = Fib), 10)
top10
# Plot variable features with and without labels
plot1 <- VariableFeaturePlot(object = Fib)
plot2 <- LabelPoints(plot = plot1, points = top10, repel = TRUE)
CombinePlots(plots = list(plot1, plot2))
```


#### Scale the data
Next, apply a linear transformation  that is a standard pre-processing step prior to dimensional reduction techniques like PCA. The `ScaleData` function:

**Highly-expressed genes might dominate downstream analyses since their expression range is much higher than most other genes.** In order to prevent this we need to;

* shift the expression of each gene, so that the mean expression across cells is 0
* scale the expression of each gene, so that the variance across cells is;

Apply the `ScaleData` function


```{r scaledata, warning = FALSE, message=FALSE}
all.genes <- rownames(x = Fib)
Fib <- ScaleData(object = Fib, features = all.genes)
```


```{r regress, cache=TRUE, warning = FALSE, message=FALSE}
# Another factor that can influence downstream analyses is cell cycle status, 
# cells that arenormally cycling but in different phases of cell cycle # might appear as separate populations.
# Seurat is preloaded with list of cell cycle markers, from Tirosh et al. 2015, 
# segregate this list into markers of G2/M phase and markers of S phase

s.genes <- cc.genes$s.genes
g2m.genes <- cc.genes$g2m.genes
Fib <- CellCycleScoring(Fib, s.features = s.genes, g2m.features = g2m.genes, set.ident = TRUE)

# We can view cell cycle scores and phase assignments with
head(Fib[[]])

# Regress out cell cycle scores during data scaling
Fib <- ScaleData(Fib, vars.to.regress = 
                   c("S.Score", "G2M.Score"),  features =rownames(Fib))
```


### Perform linear dimensional reduction

Each gene creates another dimension in our dataset, but most of those does not play a role in differenting subgroups of cells. Principal components analyses (PCA) helps us by reducing the dimensions of our data into compoments which explains most of the variation. 



```{r pca, results='hide'}
Fib <- RunPCA(object = Fib, features = VariableFeatures(object = Fib), 
              ndims.print = 10, nfeatures.print = 10)
```

#### Visualize the PCA results
```{r pca_viz, message=TRUE}
# Examine and visualize PCA results a few different ways
print(x = Fib[['pca']], dims = 1:5, nfeatures = 5)
VizDimLoadings(object = Fib, dims = 1:2, reduction = 'pca')
DimPlot(object = Fib, reduction = 'pca')
```

```{r multi-heatmap, fig.height=12, fig.width=9}
DimHeatmap(object = Fib, dims = 1:9, cells = 500, balanced = TRUE)
```


### What is the 'dimensionality' of the data?

For clustering analyses we will choose the principal components which explain most of the variation in our data. `Elbow plot` uses a ranking of principle components based on the percentage explained by each one. 

```{r elbow_plot, fig.height=6, fig.width=10}
ElbowPlot(object = Fib)
```

We did not get a clear elbow shape here, but after 10th principal component additional dimensions does not explain big amount of the variance. So we will use first 10 dimensions for the subsequent analyses.

### Cluster the cells

A graph-based clustering approach will be performed, built upon initial strategies in ([Macosko *et al*](http://www.cell.com/abstract/S0092-8674(15)00549-8)).
```{r cluster, fig.height=5, fig.width=7, message=FALSE, warning = FALSE}
Fib <- FindNeighbors(object = Fib, dims = 1:10)
Fib <- FindClusters(object = Fib, resolution = 0.5)

# Look at cluster IDs of the first 5 cells
head(Idents(Fib), 5)
```

### TSNE plot

```{r TSNE plot,message=FALSE, warning = FALSE}
Fib <- RunTSNE(object = Fib, dims = 1:6)
DimPlot(object = Fib, reduction = 'tsne', label = TRUE, label.size = 5)
```



### How to find genes that differentiate each cluster? 

We can use `FindMarkers()` function to search for cluster biomarkers.

```{r markers1, fig.height=8, fig.width=15, message = FALSE, warning = FALSE}
# Find all markers of cluster 0
cluster0.markers <- FindMarkers(object = Fib, ident.1 = 0, min.pct = 0.25)
head(x = cluster0.markers, n = 10)

# Find markers for every cluster compared to all remaining cells, report only 
# the positive ones
Fib.markers <- FindAllMarkers(object = Fib, only.pos = TRUE, min.pct = 0.25, 
                              logfc.threshold = 0.25)

# Make a table containing markers for each cluster set. 
# We will use this table to assign cell types to clusters.
Fib.markers %>% group_by(cluster) %>% top_n(n = 5, wt = avg_logFC) %>% 
  print(n = 85)
```

For more details consult to [DE vignette](http://satijalab01.nygenome.org/seurat/v3.0/de_vignette.html).


According to the table above I generated a cluster id vector to assign cell type to each cluster. I have a group NA which I could not assign a cell type, which is probably a technical artifact from cell cycle regression since the top genes expressed in this population are cell cycle related. I also found 3 subpopulations of macrophages and 2 types of mast cells in thiS patient. 

```{r tsnewithcellids, fig.height=8, fig.width=15}
# Assigning Cell ids

new.cluster.ids <- c("Macrophages(1)", "Macrophages(2)", "Macrophages(3)",
                     "AT2(1) Cells", "Cliated Cells", "Basal Cells", "Club Cells", 
                     "Dendritic Cells", "NA", "AT2(2) Cells", "Fibroblasts", 
                     "Mast(1) Cells", "Monocytes", "T/NKT Cells", "Mast(2) Cells")
names(new.cluster.ids) <- levels(Fib)
Fib <- RenameIdents(Fib, new.cluster.ids)
DimPlot(Fib, reduction = "tsne", label = TRUE, label.size = 8, pt.size = 1.0) + NoLegend()
```



```{r differentfeautures}

# Let's look at the expression of a few genes of interest accross different subtypes
VlnPlot(Fib, features = c("SPP1", "CHIT1"))

FeaturePlot(Fib, features = c("SPP1", "APOE", "SCGB3A2", "C9orf24", "S100A2", "BPIFB1", "S100B", "COL3A1", "FCN1"))
```

### What cells `Biotech` Companies are targeting in lung fibrosis?

Novel therapies in pulmonary fibrosis are urgently needed. Let's look at few promising drugs currently in clinical trials.

#### Fibrinogen

Fibrinoge's lead compound pamrevlumab blocks CTGF. Let's have a look which cells produce it.  

```{r Fibrinogen}

VlnPlot(Fib, features =c("CTGF"))
        
FeaturePlot(Fib, features = c("CTGF"), pt.size = 1)
```
Main source for the `CTGF` production here seems the ``Ciliated cells.``

#### Galapagos

Galapagos GLPG1690 has entered Phase 3 clinical trials and it targets Autotaxin (ENPP2). 

```{r GLPG1690}

VlnPlot(Fib, features =  c("ENPP2"))

FeaturePlot(Fib, features =  c("ENPP2"),  pt.size = 1)
```
We see a low expression of `ENPP` mainly in Macrophages, Dendritic cells and Fibroblasts. 

The other clinical candidate of Galapagos is GLPG1205 which targets GPR84. It has entered Phase 2 clinical trials.
```{r GLPG1205}

VlnPlot(Fib, features =  c("GPR84"))

FeaturePlot(Fib, features =  c("GPR84"), pt.size = 1)
```

`GPR84` has very low expression throughout the lung cells of this patient. The gene is mainly expressed in Type 1 Macrophages. 

#### Bristol-Myers-Squibb: `BMS-986020`
BMS-986020 is an anti-fibrotic drug being developed by Bristol-Myers Squibb, and is a lysophosphatidic acid (LPA) receptor antagonist. 

Let's look at the expression of a LPAR1.
```{r LPAR1}


VlnPlot(Fib, features = c("LPAR1"))

FeaturePlot(Fib, features =c("LPAR1"),  pt.size = 1)
```
It is expressed at low levels but consistently on many types of lung cells.

#### Roche `nintedanib`

Nintedanib is one of the two new medicines approved for IPF patients. It targets multiple receptor tyrosine kinase receptors such as VEGF, FGF and PDGF.
```{r nintedanib}

VlnPlot(Fib, features = c("VEGFA", "PDGFA"))

FeaturePlot(Fib, features =  c("VEGFA", "PDGFA"), pt.size = 1)
```

VEGFA, PDGFA is highly expressed in multiple types of cells in the lung microenvironment.

### Final thoughts

Seurat package allowed us to cluster different types of lung cells so that we can visualize their specific gene expression. We used those plots to see which cells express the drug targets in clinical trials or in use.  
Although changes associated with fibrosis are well defined the causal factors are not known. Reflecting this we saw that those **drugs target many different components of the lung microenvironment.** The conquest is on and we hope promising therapies arrive soon.

