<!DOCTYPE html>
<html lang="en-us">

<head>

  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="generator" content="Source Themes Academic 4.5.0">

  

  
  
  
  
  
    
    
    
  
  

  <meta name="author" content="Serdar Korur">

  
  
  
    
  
  <meta name="description" content="Advances in microfluidic technologies enabled us to barcode single cells in lipid droplets and to resolve genomes of individual cells from a sequencing mixture (e.g, 10X Genomics). By using Single cell RNA sequencing (scRNA-seq) we can discover rare cell populations and genes that are specifically acting in those. Potential is high and the list of publications growing daily.
If you are a scientist in a biotech exploring novel targets those might be a great source to gather specific information.">

  
  <link rel="alternate" hreflang="en-us" href="/2019/09/17/genomics-at-superresolution/">

  


  
  
  
  <meta name="theme-color" content="#2962ff">
  

  
  
  
  
    
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/academicons/1.8.6/css/academicons.min.css" integrity="sha256-uFVgMKfistnJAfoCUQigIl+JfUaP47GrRKjf6CTPVmw=" crossorigin="anonymous">
    <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.6.0/css/all.css" integrity="sha384-aOkxzJ5uQz7WBObEZcHvV5JvRW3TUc2rNPA7pe3AwnsUohiw1Vj2Rgx2KSOkF5+h" crossorigin="anonymous">
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.2.5/jquery.fancybox.min.css" integrity="sha256-ygkqlh3CYSUri3LhQxzdcm0n1EQvH2Y+U5S2idbLtxs=" crossorigin="anonymous">

    
    
    
      
    
    
      
      
        
          <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.15.6/styles/github.min.css" crossorigin="anonymous" title="hl-light">
          <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.15.6/styles/dracula.min.css" crossorigin="anonymous" title="hl-dark" disabled>
        
      
    

    
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/leaflet/1.2.0/leaflet.css" integrity="sha512-M2wvCLH6DSRazYeZRIm1JnYyh22purTM+FDB5CsyxtQJYeKq83arPe5wgbNmcFXGqiSH2XR8dT/fJISVA1r/zQ==" crossorigin="anonymous">
    

    

  

  
  
  
  <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Montserrat:400,700%7CRoboto:400,400italic,700%7CRoboto+Mono&display=swap">
  

  
  
  
  <link rel="stylesheet" href="/css/academic.min.54abc5f3f1ddc380a34d4103fd11d77a.css">

  

  
  
    <script>
      window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;
      ga('create', 'UA-148414815-1', 'auto');
      
      ga('require', 'eventTracker');
      ga('require', 'outboundLinkTracker');
      ga('require', 'urlChangeTracker');
      ga('send', 'pageview');
    </script>
    <script async src="https://www.google-analytics.com/analytics.js"></script>
    
    <script async src="https://cdnjs.cloudflare.com/ajax/libs/autotrack/2.4.1/autotrack.js" integrity="sha512-HUmooslVKj4m6OBu0OgzjXXr+QuFYy/k7eLI5jdeEy/F4RSgMn6XRWRGkFi5IFaFgy7uFTkegp3Z0XnJf3Jq+g==" crossorigin="anonymous"></script>
    
  
  

  

  <link rel="manifest" href="/index.webmanifest">
  <link rel="icon" type="image/png" href="/img/icon-32.png">
  <link rel="apple-touch-icon" type="image/png" href="/img/icon-192.png">

  <link rel="canonical" href="/2019/09/17/genomics-at-superresolution/">

  
  
  
  
    
    
  
  
  <meta property="twitter:card" content="summary">
  
  <meta property="og:site_name" content="SERDAR KORUR">
  <meta property="og:url" content="/2019/09/17/genomics-at-superresolution/">
  <meta property="og:title" content="Genomics at superresolution: Mapping Drug targets on single cell resolution in Fibrosis | SERDAR KORUR">
  <meta property="og:description" content="Advances in microfluidic technologies enabled us to barcode single cells in lipid droplets and to resolve genomes of individual cells from a sequencing mixture (e.g, 10X Genomics). By using Single cell RNA sequencing (scRNA-seq) we can discover rare cell populations and genes that are specifically acting in those. Potential is high and the list of publications growing daily.
If you are a scientist in a biotech exploring novel targets those might be a great source to gather specific information."><meta property="og:image" content="/img/icon-192.png">
  <meta property="twitter:image" content="/img/icon-192.png"><meta property="og:locale" content="en-us">
  
    
      <meta property="article:published_time" content="2019-09-17T00:00:00&#43;00:00">
    
    <meta property="article:modified_time" content="2019-09-17T21:23:26&#43;02:00">
  

  


    






  





  





  





<script type="application/ld+json">
{
  "@context": "https://schema.org",
  "@type": "BlogPosting",
  "mainEntityOfPage": {
    "@type": "WebPage",
    "@id": "/2019/09/17/genomics-at-superresolution/"
  },
  "headline": "Genomics at superresolution: Mapping Drug targets on single cell resolution in Fibrosis",
  
  "datePublished": "2019-09-17T00:00:00Z",
  "dateModified": "2019-09-17T21:23:26+02:00",
  
  "author": {
    "@type": "Person",
    "name": "Serdar Korur"
  },
  
  "publisher": {
    "@type": "Organization",
    "name": "SERDAR KORUR",
    "logo": {
      "@type": "ImageObject",
      "url": "/img/icon-512.png"
    }
  },
  "description": "Advances in microfluidic technologies enabled us to barcode single cells in lipid droplets and to resolve genomes of individual cells from a sequencing mixture (e.g, 10X Genomics). By using Single cell RNA sequencing (scRNA-seq) we can discover rare cell populations and genes that are specifically acting in those. Potential is high and the list of publications growing daily.\n\rIf you are a scientist in a biotech exploring novel targets those might be a great source to gather specific information."
}
</script>

  

  


  


  





  <title>Genomics at superresolution: Mapping Drug targets on single cell resolution in Fibrosis | SERDAR KORUR</title>

</head>

<body id="top" data-spy="scroll" data-offset="70" data-target="#TableOfContents" >

  <aside class="search-results" id="search">
  <div class="container">
    <section class="search-header">

      <div class="row no-gutters justify-content-between mb-3">
        <div class="col-6">
          <h1>Search</h1>
        </div>
        <div class="col-6 col-search-close">
          <a class="js-search" href="#"><i class="fas fa-times-circle text-muted" aria-hidden="true"></i></a>
        </div>
      </div>

      <div id="search-box">
        
        <input name="q" id="search-query" placeholder="Search..." autocapitalize="off"
        autocomplete="off" autocorrect="off" spellcheck="false" type="search">
        
      </div>

    </section>
    <section class="section-search-results">

      <div id="search-hits">
        
      </div>

    </section>
  </div>
</aside>


  
<nav class="navbar navbar-light fixed-top navbar-expand-lg py-0 compensate-for-scrollbar" id="navbar-main">
  <div class="container">

    
      <a class="navbar-brand" href="/">SERDAR KORUR</a>
      
      <button type="button" class="navbar-toggler" data-toggle="collapse"
              data-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
        <span><i class="fas fa-bars"></i></span>
      </button>
      

    
    <div class="collapse navbar-collapse" id="navbar">

      
      
      <ul class="navbar-nav mr-auto">
        

        

        
        
        
          
        

        
        
        
        
        
        
          
          
          
            
          
          
        

        <li class="nav-item">
          <a class="nav-link " href="/#about"><span>Home</span></a>
        </li>

        
        

        

        
        
        
          
        

        
        
        
        
        
        
          
          
          
            
          
          
        

        <li class="nav-item">
          <a class="nav-link " href="/#posts"><span>Posts</span></a>
        </li>

        
        

        

        
        
        
          
        

        
        
        
        
        
        
          
          
          
            
          
          
        

        <li class="nav-item">
          <a class="nav-link " href="/#contact"><span></span></a>
        </li>

        
        

      
      </ul>
      <ul class="navbar-nav ml-auto">
      

        

        
        <li class="nav-item">
          <a class="nav-link js-search" href="#"><i class="fas fa-search" aria-hidden="true"></i></a>
        </li>
        

        

        
        <li class="nav-item">
          <a class="nav-link js-dark-toggle" href="#"><i class="fas fa-moon" aria-hidden="true"></i></a>
        </li>
        

      </ul>

    </div>
  </div>
</nav>


  <article class="article">

  












  

  
  
  
<div class="article-container pt-3">
  <h1>Genomics at superresolution: Mapping Drug targets on single cell resolution in Fibrosis</h1>

  

  
    



<div class="article-metadata">

  
  

  
  <span class="article-date">
    
    
      
          Last updated on
      
    
    Sep 17, 2019
  </span>
  

  

  
  <span class="middot-divider"></span>
  <span class="article-reading-time">
    14 min read
  </span>
  

  
  
  
  <span class="middot-divider"></span>
  <a href="/2019/09/17/genomics-at-superresolution/#disqus_thread"></a>
  

  
  

  
    
<div class="share-box" aria-hidden="true">
  <ul class="share">
    
      
      
      
        
      
      
      
      <li>
        <a href="https://twitter.com/intent/tweet?url=/2019/09/17/genomics-at-superresolution/&amp;text=Genomics%20at%20superresolution:%20Mapping%20Drug%20targets%20on%20single%20cell%20resolution%20in%20Fibrosis" target="_blank" rel="noopener" class="share-btn-twitter">
          <i class="fab fa-twitter"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="https://www.facebook.com/sharer.php?u=/2019/09/17/genomics-at-superresolution/&amp;t=Genomics%20at%20superresolution:%20Mapping%20Drug%20targets%20on%20single%20cell%20resolution%20in%20Fibrosis" target="_blank" rel="noopener" class="share-btn-facebook">
          <i class="fab fa-facebook-f"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="mailto:?subject=Genomics%20at%20superresolution:%20Mapping%20Drug%20targets%20on%20single%20cell%20resolution%20in%20Fibrosis&amp;body=/2019/09/17/genomics-at-superresolution/" target="_blank" rel="noopener" class="share-btn-email">
          <i class="fas fa-envelope"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="https://www.linkedin.com/shareArticle?url=/2019/09/17/genomics-at-superresolution/&amp;title=Genomics%20at%20superresolution:%20Mapping%20Drug%20targets%20on%20single%20cell%20resolution%20in%20Fibrosis" target="_blank" rel="noopener" class="share-btn-linkedin">
          <i class="fab fa-linkedin-in"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="https://web.whatsapp.com/send?text=Genomics%20at%20superresolution:%20Mapping%20Drug%20targets%20on%20single%20cell%20resolution%20in%20Fibrosis%20/2019/09/17/genomics-at-superresolution/" target="_blank" rel="noopener" class="share-btn-whatsapp">
          <i class="fab fa-whatsapp"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="https://service.weibo.com/share/share.php?url=/2019/09/17/genomics-at-superresolution/&amp;title=Genomics%20at%20superresolution:%20Mapping%20Drug%20targets%20on%20single%20cell%20resolution%20in%20Fibrosis" target="_blank" rel="noopener" class="share-btn-weibo">
          <i class="fab fa-weibo"></i>
        </a>
      </li>
    
  </ul>
</div>


  

</div>

    














  
</div>



  <div class="article-container">

    <div class="article-style">
      


<p>Advances in <strong>microfluidic technologies</strong> enabled us to barcode single cells in lipid droplets and to <strong>resolve genomes of individual cells</strong> from a sequencing mixture (e.g, <a href="https://www.10xgenomics.com/">10X Genomics</a>). By using <strong>Single cell RNA sequencing (scRNA-seq)</strong> we can discover <strong>rare cell populations</strong> and genes that are specifically acting in those. Potential is high and the list of publications growing daily.</p>
<ul>
<li>If you are a scientist in a biotech exploring novel targets those might be a great source to <strong>gather specific information</strong>.</li>
</ul>
<p><a href="https://cran.r-project.org/web/packages/Seurat/index.html">Seurat</a> package is a great tool for digging into single cell datasets. It will open you access beyond what is in the publications. You can find many tutorials in their <a href="https://satijalab.org/seurat/">website</a>.</p>
<p>Here, I will focus on a recent <a href="https://www.atsjournals.org/doi/pdf/10.1164/rccm.201712-2410OC">paper</a> which explored transcriptome of lung cells from Pulmonary fibrosis patients by scRNAseq.</p>
<ul>
<li><strong>Pulmonary fibrosis</strong> is a progressive scarring of the lung tissue leading to death within 3-4 years.</li>
<li><strong>Current therapies does not increase the survival</strong></li>
<li>Many Biotech companies are developing <strong>novel drugs</strong></li>
</ul>
<p><div class="alert alert-note">
  <div>
    What types of cells those drugs are acting on?
  </div>
</div>
</p>
<p>Our workflow will be in three steps</p>
<ol style="list-style-type: decimal">
<li>Basic Seurat workflow</li>
<li>Identify cell types on a tSNE plot</li>
<li>Visualize drug targets on single cell plots</li>
</ol>
<p>I picked up scRNA-seq data from a patient with Polymyositis associated intersitial lung disease.</p>
<p>—&gt; <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM3489196">Download Single Cell RNA seq data</a></p>
<div id="seurat-workflow" class="section level3">
<h3>Seurat Workflow</h3>
<p>Load the packages that will be used.</p>
<pre class="r"><code>library(Seurat)
library(dplyr)
library(ggplot2)</code></pre>
</div>
<div id="importing-the-h5-file" class="section level3">
<h3>Importing the h5 file</h3>
<p>Here the data comes as an <strong>h5 file.</strong> But no worries, we can handle that with <strong>Read10X_h5</strong> function in Seurat. Since the file is directly under my working directory I can apply <code>Read10X_h5("Filename.h5")</code></p>
<pre class="r"><code>MyoILD_01 &lt;- Read10X_h5(&quot;posts_data/GSM3489196_Myositis-ILD_01_filtered_gene_bc_matrices_h5.h5&quot;)
Fib &lt;- CreateSeuratObject(counts = MyoILD_01, min.cells = 3, project= &quot;FightFibrosis&quot;, min.features = 200)
Fib</code></pre>
<pre><code>## An object of class Seurat 
## 20246 features across 7163 samples within 1 assay 
## Active assay: RNA (20246 features)</code></pre>
<p>Our Data matrix now contains <strong>20246</strong> genes and <strong>7163</strong> cells</p>
</div>
<div id="a-quick-look-at-the-raw-count-data" class="section level3">
<h3>A Quick look at the raw count data</h3>
<pre class="r"><code># Lets examine a few genes in the first thirty cells
MyoILD_01[c(&quot;SPP1&quot;,&quot;TGFB1&quot;,&quot;CCL2&quot;), 1:30]</code></pre>
<pre><code>## 3 x 30 sparse Matrix of class &quot;dgCMatrix&quot;
##                                                                    
## SPP1  1 . . 2 . . . 33 . . 1 . . . . . . . . . . . . . . . 9 . . 20
## TGFB1 . . . . 1 . .  . . . . . . . 1 . 3 . . . . . . . 1 2 . . .  .
## CCL2  5 . . 2 . . .  . . . . . . . . . . . . . . . . . . . . . .  .</code></pre>
</div>
<div id="standard-pre-processing-workflow-for-scrna-seq-data" class="section level3">
<h3>Standard pre-processing workflow for scRNA-seq data</h3>
<p>Pre-processing involves filtration of cells based on Quality control metrics (e.g .mitochondrial contamination, Coverage). We will proceed by normalization and identification of the highly variable features then at the end we will scale the data.</p>
<div id="qc-and-selecting-cells-for-further-analysis" class="section level4">
<h4>QC and selecting cells for further analysis</h4>
<pre class="r"><code># Let&#39;s add Mitochondrial stats to Fib data. [[ operator can add 
# columns to object metadata. 

Fib[[&quot;percent.mt&quot;]] &lt;- PercentageFeatureSet(object = Fib, pattern = &quot;^MT-&quot;)</code></pre>
<pre class="r"><code># Quick look at the Quality control metrics for the first 5 cells
head(x = Fib@meta.data, 5)</code></pre>
<pre><code>##                       orig.ident nCount_RNA nFeature_RNA percent.mt
## AAACCTGAGATCCCGC-1 FightFibrosis       2257         1010   4.386354
## AAACCTGAGCAGCCTC-1 FightFibrosis       3367          713  66.914167
## AAACCTGAGCCAGAAC-1 FightFibrosis       7354         1943   3.929834
## AAACCTGAGGGATGGG-1 FightFibrosis       9323         2786   8.430763
## AAACCTGAGTCGTACT-1 FightFibrosis       2937         1125   6.775621</code></pre>
</div>
<div id="violin-plots-to-visualize-qc-metrics" class="section level4">
<h4>Violin plots to visualize QC metrics</h4>
<pre class="r"><code>#Visualize QC metrics as a violin plot
VlnPlot(object = Fib, features = c(&quot;nFeature_RNA&quot;, &quot;nCount_RNA&quot;, &quot;percent.mt&quot;),
        ncol = 3)</code></pre>
<p><img src="/post/2019-09-17-genomics-at-superresolution-advances-in-sequencing-technologies_files/figure-html/qc2-1.png" width="1248" /></p>
<pre class="r"><code># We will use FeatureScatter to visualize feature-feature relationships like 
# RNA counts or percentage of mitochondiral contamination

plot1 &lt;- FeatureScatter(object = Fib, feature1 = &quot;nCount_RNA&quot;, feature2 = &quot;percent.mt&quot;) 
plot2 &lt;- FeatureScatter(object = Fib, feature1 = &quot;nCount_RNA&quot;, feature2 = &quot;nFeature_RNA&quot;) 
CombinePlots(plots = list(plot1,plot2))</code></pre>
<p><img src="/post/2019-09-17-genomics-at-superresolution-advances-in-sequencing-technologies_files/figure-html/qc2-2.png" width="1248" /></p>
<pre class="r"><code># As you see on the left plot cells with high percentage of mitochondrial genes
# have very low numbers of RNA indicating that they are low quality/dead cells.
# Let&#39;s remove them.
Fib &lt;- subset(x = Fib, subset = nFeature_RNA &gt; 200 &amp; nFeature_RNA &lt; 4000 &amp; 
              percent.mt &lt; 12.5)</code></pre>
</div>
<div id="normalize-the-data" class="section level4">
<h4>Normalize the data</h4>
<p>Log normalization helps to reduce the influences of the outliers.
Normalized values will be stored in <code>Fib[["RNA"]]@data</code>.</p>
<pre class="r"><code>Fib &lt;- NormalizeData(object = Fib, normalization.method = &quot;LogNormalize&quot;, 
                     scale.factor = 1e4)</code></pre>
</div>
<div id="identify-highly-variable-features" class="section level4">
<h4>Identify highly variable features</h4>
<p>Let’s calculate the features that exhibit high cell-to-cell variation in the dataset. Focusing on those genes will help to highlight biological signal.</p>
<pre class="r"><code>Fib &lt;- FindVariableFeatures(object = Fib, selection.method = &#39;vst&#39;,  nfeatures = 2000)

# Identify the 10 most highly variable genes
top10 &lt;- head(x = VariableFeatures(object = Fib), 10)
top10</code></pre>
<pre><code>##  [1] &quot;SCGB1A1&quot; &quot;SCGB3A1&quot; &quot;SFTPC&quot;   &quot;SCGB3A2&quot; &quot;HBB&quot;     &quot;PLA2G2A&quot; &quot;MT1G&quot;   
##  [8] &quot;TPSB2&quot;   &quot;FABP4&quot;   &quot;MT1H&quot;</code></pre>
<pre class="r"><code># Plot variable features with and without labels
plot1 &lt;- VariableFeaturePlot(object = Fib)
plot2 &lt;- LabelPoints(plot = plot1, points = top10, repel = TRUE)
CombinePlots(plots = list(plot1, plot2))</code></pre>
<p><img src="/post/2019-09-17-genomics-at-superresolution-advances-in-sequencing-technologies_files/figure-html/var_features-1.png" width="1056" /></p>
</div>
<div id="scale-the-data" class="section level4">
<h4>Scale the data</h4>
<p>Next, apply a linear transformation that is a standard pre-processing step prior to dimensional reduction techniques like PCA. The <code>ScaleData</code> function:</p>
<p><strong>Highly-expressed genes might dominate downstream analyses since their expression range is much higher than most other genes.</strong> In order to prevent this we need to;</p>
<ul>
<li>shift the expression of each gene, so that the mean expression across cells is 0</li>
<li>scale the expression of each gene, so that the variance across cells is;</li>
</ul>
<p>Apply the <code>ScaleData</code> function</p>
<pre class="r"><code>all.genes &lt;- rownames(x = Fib)
Fib &lt;- ScaleData(object = Fib, features = all.genes)</code></pre>
<pre class="r"><code># Another factor that can influence downstream analyses is cell cycle status, 
# cells that arenormally cycling but in different phases of cell cycle # might appear as separate populations.
# Seurat is preloaded with list of cell cycle markers, from Tirosh et al. 2015, 
# segregate this list into markers of G2/M phase and markers of S phase

s.genes &lt;- cc.genes$s.genes
g2m.genes &lt;- cc.genes$g2m.genes
Fib &lt;- CellCycleScoring(Fib, s.features = s.genes, g2m.features = g2m.genes, set.ident = TRUE)

# We can view cell cycle scores and phase assignments with
head(Fib[[]])</code></pre>
<pre><code>##                       orig.ident nCount_RNA nFeature_RNA percent.mt
## AAACCTGAGATCCCGC-1 FightFibrosis       2257         1010   4.386354
## AAACCTGAGCCAGAAC-1 FightFibrosis       7354         1943   3.929834
## AAACCTGAGGGATGGG-1 FightFibrosis       9323         2786   8.430763
## AAACCTGAGTCGTACT-1 FightFibrosis       2937         1125   6.775621
## AAACCTGAGTCGTTTG-1 FightFibrosis       5447         1719   4.149073
## AAACCTGCAAGTCTAC-1 FightFibrosis       6453         1836   2.773904
##                         S.Score    G2M.Score Phase     old.ident
## AAACCTGAGATCCCGC-1  0.006739441 -0.016659228     S FightFibrosis
## AAACCTGAGCCAGAAC-1 -0.040762255 -0.010151464    G1 FightFibrosis
## AAACCTGAGGGATGGG-1 -0.048223904 -0.025580453    G1 FightFibrosis
## AAACCTGAGTCGTACT-1  0.076114887  0.009717348     S FightFibrosis
## AAACCTGAGTCGTTTG-1 -0.045171005 -0.053868788    G1 FightFibrosis
## AAACCTGCAAGTCTAC-1  0.003354484 -0.009033119     S FightFibrosis</code></pre>
<pre class="r"><code># Regress out cell cycle scores during data scaling
Fib &lt;- ScaleData(Fib, vars.to.regress = 
                   c(&quot;S.Score&quot;, &quot;G2M.Score&quot;),  features =rownames(Fib))</code></pre>
</div>
</div>
<div id="perform-linear-dimensional-reduction" class="section level3">
<h3>Perform linear dimensional reduction</h3>
<p>Each gene creates another dimension in our dataset, but most of those does not play a role in differenting subgroups of cells. Principal components analyses (PCA) helps us by reducing the dimensions of our data into compoments which explains most of the variation.</p>
<pre class="r"><code>Fib &lt;- RunPCA(object = Fib, features = VariableFeatures(object = Fib), 
              ndims.print = 10, nfeatures.print = 10)</code></pre>
<pre><code>## PC_ 10 
## Positive:  VCAN, MCEMP1, S100A8, FCN1, CD2, IL32, S100A6, TIMP1, CCL5, CD3D 
## Negative:  CHIT1, A2M, CHCHD6, AC079767.4, SEPP1, ALOX15B, GPNMB, DNASE2B, LIPA, SLC40A1</code></pre>
<div id="visualize-the-pca-results" class="section level4">
<h4>Visualize the PCA results</h4>
<pre class="r"><code># Examine and visualize PCA results a few different ways
print(x = Fib[[&#39;pca&#39;]], dims = 1:5, nfeatures = 5)</code></pre>
<pre><code>## PC_ 1 
## Positive:  TMSB4X, TYROBP, FTL, FCER1G, VIM 
## Negative:  RSPH1, C9orf24, TMEM190, C20orf85, FAM183A 
## PC_ 2 
## Positive:  FTH1, CTSS, TYROBP, B2M, FCER1G 
## Negative:  GPRC5A, SFTPB, CEACAM6, HOPX, MUC1 
## PC_ 3 
## Positive:  PLA2G2A, GFPT2, MEDAG, COL3A1, COL1A2 
## Negative:  SFTPB, SFTPA2, SFTA2, SFTPA1, SLC34A2 
## PC_ 4 
## Positive:  SFTPC, NAPSA, SFTPD, LRRK2, LAMP3 
## Negative:  S100A2, KRT19, KRT5, AKR1C1, BPIFB1 
## PC_ 5 
## Positive:  CPA3, TPSAB1, TPSB2, MS4A2, GATA2 
## Negative:  S100A6, CSTB, S100A10, GCHFR, TXN</code></pre>
<pre class="r"><code>VizDimLoadings(object = Fib, dims = 1:2, reduction = &#39;pca&#39;)</code></pre>
<p><img src="/post/2019-09-17-genomics-at-superresolution-advances-in-sequencing-technologies_files/figure-html/pca_viz-1.png" width="672" /></p>
<pre class="r"><code>DimPlot(object = Fib, reduction = &#39;pca&#39;)</code></pre>
<p><img src="/post/2019-09-17-genomics-at-superresolution-advances-in-sequencing-technologies_files/figure-html/pca_viz-2.png" width="672" /></p>
<pre class="r"><code>DimHeatmap(object = Fib, dims = 1:9, cells = 500, balanced = TRUE)</code></pre>
<p><img src="/post/2019-09-17-genomics-at-superresolution-advances-in-sequencing-technologies_files/figure-html/multi-heatmap-1.png" width="864" /></p>
</div>
</div>
<div id="what-is-the-dimensionality-of-the-data" class="section level3">
<h3>What is the ‘dimensionality’ of the data?</h3>
<p>For clustering analyses we will choose the principal components which explain most of the variation in our data. <code>Elbow plot</code> uses a ranking of principle components based on the percentage explained by each one.</p>
<pre class="r"><code>ElbowPlot(object = Fib)</code></pre>
<p><img src="/post/2019-09-17-genomics-at-superresolution-advances-in-sequencing-technologies_files/figure-html/elbow_plot-1.png" width="960" /></p>
<p>We did not get a clear elbow shape here, but after 10th principal component additional dimensions does not explain big amount of the variance. So we will use first 10 dimensions for the subsequent analyses.</p>
</div>
<div id="cluster-the-cells" class="section level3">
<h3>Cluster the cells</h3>
<p>A graph-based clustering approach will be performed, built upon initial strategies in (<a href="http://www.cell.com/abstract/S0092-8674(15)00549-8">Macosko <em>et al</em></a>).</p>
<pre class="r"><code>Fib &lt;- FindNeighbors(object = Fib, dims = 1:10)
Fib &lt;- FindClusters(object = Fib, resolution = 0.5)</code></pre>
<pre><code>## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck
## 
## Number of nodes: 6320
## Number of edges: 196713
## 
## Running Louvain algorithm...
## Maximum modularity in 10 random starts: 0.8872
## Number of communities: 15
## Elapsed time: 0 seconds</code></pre>
<pre class="r"><code># Look at cluster IDs of the first 5 cells
head(Idents(Fib), 5)</code></pre>
<pre><code>## AAACCTGAGATCCCGC-1 AAACCTGAGCCAGAAC-1 AAACCTGAGGGATGGG-1 
##                  2                  2                  8 
## AAACCTGAGTCGTACT-1 AAACCTGAGTCGTTTG-1 
##                  0                 10 
## Levels: 0 1 2 3 4 5 6 7 8 9 10 11 12 13 14</code></pre>
</div>
<div id="tsne-plot" class="section level3">
<h3>TSNE plot</h3>
<pre class="r"><code>Fib &lt;- RunTSNE(object = Fib, dims = 1:6)
DimPlot(object = Fib, reduction = &#39;tsne&#39;, label = TRUE, label.size = 5)</code></pre>
<p><img src="/post/2019-09-17-genomics-at-superresolution-advances-in-sequencing-technologies_files/figure-html/TSNE%20plot-1.png" width="672" /></p>
</div>
<div id="how-to-find-genes-that-differentiate-each-cluster" class="section level3">
<h3>How to find genes that differentiate each cluster?</h3>
<p>We can use <code>FindMarkers()</code> function to search for cluster biomarkers.</p>
<pre class="r"><code># Find all markers of cluster 0
cluster0.markers &lt;- FindMarkers(object = Fib, ident.1 = 0, min.pct = 0.25)
head(x = cluster0.markers, n = 10)</code></pre>
<pre><code>##          p_val avg_logFC pct.1 pct.2 p_val_adj
## GPNMB        0 1.0814399 0.947 0.450         0
## CSTB         0 1.0377160 0.994 0.892         0
## FTL          0 1.0160962 1.000 0.998         0
## APOC1        0 0.9919471 0.991 0.819         0
## SH3BGRL3     0 0.9252345 0.990 0.798         0
## CTSD         0 0.9105122 0.995 0.836         0
## FTH1         0 0.8873914 1.000 0.997         0
## PSAP         0 0.8835586 1.000 0.876         0
## CTSB         0 0.8773872 0.986 0.752         0
## LGALS1       0 0.8622147 0.992 0.637         0</code></pre>
<pre class="r"><code># Find markers for every cluster compared to all remaining cells, report only 
# the positive ones
Fib.markers &lt;- FindAllMarkers(object = Fib, only.pos = TRUE, min.pct = 0.25, 
                              logfc.threshold = 0.25)

# Make a table containing markers for each cluster set. 
# We will use this table to assign cell types to clusters.
Fib.markers %&gt;% group_by(cluster) %&gt;% top_n(n = 5, wt = avg_logFC) %&gt;% 
  print(n = 85)</code></pre>
<pre><code>## # A tibble: 75 x 7
## # Groups:   cluster [15]
##        p_val avg_logFC pct.1 pct.2 p_val_adj cluster gene    
##        &lt;dbl&gt;     &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;     &lt;dbl&gt; &lt;fct&gt;   &lt;chr&gt;   
##  1 0.            1.08  0.947 0.45  0.        0       GPNMB   
##  2 2.60e-293     1.13  0.935 0.582 5.27e-289 0       FABP5   
##  3 8.54e-289     1.12  0.814 0.365 1.73e-284 0       CTSL    
##  4 3.43e-188     1.43  0.46  0.133 6.95e-184 0       SPP1    
##  5 2.78e-110     1.37  0.288 0.078 5.64e-106 0       CCL2    
##  6 2.69e-231     1.33  0.996 0.658 5.45e-227 1       APOE    
##  7 1.91e-204     1.02  0.926 0.403 3.86e-200 1       MS4A6A  
##  8 1.35e-174     0.918 0.932 0.423 2.74e-170 1       C1QC    
##  9 1.59e-119     1.07  0.86  0.457 3.22e-115 1       CCL18   
## 10 1.56e- 56     0.916 0.576 0.311 3.15e- 52 1       SEPP1   
## 11 0.            2.26  0.663 0.108 0.        2       FABP4   
## 12 0.            1.43  0.846 0.165 0.        2       MCEMP1  
## 13 5.19e-222     1.24  0.969 0.496 1.05e-217 2       GCHFR   
## 14 4.41e-218     1.36  0.98  0.505 8.93e-214 2       C1QA    
## 15 4.43e-210     1.35  0.967 0.483 8.97e-206 2       C1QB    
## 16 0.            3.18  0.964 0.339 0.        3       SCGB3A2 
## 17 0.            2.01  0.995 0.376 0.        3       SFTPB   
## 18 0.            1.73  0.943 0.188 0.        3       CEACAM6 
## 19 2.27e-273     2.23  0.929 0.364 4.59e-269 3       SFTPA2  
## 20 8.38e-271     1.72  0.813 0.203 1.70e-266 3       SFTPA1  
## 21 0.            2.70  0.896 0.035 0.        4       C9orf24 
## 22 0.            2.46  0.87  0.026 0.        4       TMEM190 
## 23 0.            2.44  0.892 0.026 0.        4       RSPH1   
## 24 0.            2.42  0.868 0.025 0.        4       C20orf85
## 25 0.            2.39  0.907 0.104 0.        4       TPPP3   
## 26 0.            2.76  0.975 0.227 0.        5       S100A2  
## 27 0.            2.19  0.797 0.083 0.        5       KRT15   
## 28 0.            1.72  0.684 0.048 0.        5       KRT5    
## 29 6.72e-278     1.93  0.471 0.035 1.36e-273 5       MMP1    
## 30 9.96e-220     1.75  0.925 0.319 2.02e-215 5       AQP3    
## 31 0.            2.45  0.737 0.077 0.        6       BPIFB1  
## 32 6.07e-262     2.14  0.984 0.305 1.23e-257 6       WFDC2   
## 33 2.03e-221     2.21  0.982 0.385 4.11e-217 6       SLPI    
## 34 1.17e-163     3.21  0.948 0.624 2.36e-159 6       SCGB3A1 
## 35 8.07e-152     2.91  0.797 0.293 1.63e-147 6       SCGB1A1 
## 36 0.            2.57  0.797 0.072 0.        7       S100B   
## 37 9.62e-289     1.76  0.829 0.156 1.95e-284 7       FAM26F  
## 38 2.50e-224     2.18  1     0.728 5.06e-220 7       HLA-DPB1
## 39 2.51e-213     1.99  0.992 0.658 5.07e-209 7       HLA-DPA1
## 40 1.75e-191     1.71  0.955 0.449 3.55e-187 7       HLA-DQA1
## 41 4.11e- 50     1.01  0.546 0.231 8.31e- 46 8       STMN1   
## 42 1.29e- 41     0.917 0.819 0.529 2.60e- 37 8       TUBA1B  
## 43 1.29e- 40     0.915 0.807 0.517 2.61e- 36 8       TUBB    
## 44 1.32e- 29     0.973 0.825 0.615 2.67e- 25 8       H2AFZ   
## 45 5.14e- 29     1.16  0.77  0.503 1.04e- 24 8       HIST1H4C
## 46 0.            2.28  0.984 0.113 0.        9       NAPSA   
## 47 3.65e-299     2.20  0.984 0.159 7.39e-295 9       SFTA2   
## 48 8.60e-252     2.99  1     0.226 1.74e-247 9       SFTPA1  
## 49 3.66e-220     4.21  0.961 0.265 7.42e-216 9       SFTPC   
## 50 4.19e-168     2.33  0.992 0.39  8.47e-164 9       SFTPA2  
## 51 0.            2.61  0.570 0.01  0.        10      COL3A1  
## 52 0.            2.56  0.604 0.005 0.        10      SPARCL1 
## 53 0.            2.42  0.456 0.005 0.        10      PLA2G2A 
## 54 0.            2.40  0.544 0.005 0.        10      COL1A2  
## 55 1.09e-278     2.57  0.537 0.02  2.20e-274 10      COL1A1  
## 56 0.            4.23  0.993 0.035 0.        11      TPSB2   
## 57 0.            3.74  0.993 0.02  0.        11      TPSAB1  
## 58 0.            3.32  1     0.02  0.        11      CPA3    
## 59 0.            2.78  0.911 0.017 0.        11      MS4A2   
## 60 0.            2.33  0.856 0.016 0.        11      KIT     
## 61 1.98e-262     2.08  0.624 0.023 4.01e-258 12      FCN1    
## 62 5.17e- 70     1.95  0.475 0.059 1.05e- 65 12      IL1R2   
## 63 2.47e- 66     1.98  0.634 0.119 5.01e- 62 12      VCAN    
## 64 4.37e- 58     3.03  0.812 0.258 8.85e- 54 12      S100A8  
## 65 2.45e- 46     2.67  0.901 0.466 4.96e- 42 12      S100A9  
## 66 2.37e- 68     2.24  0.593 0.055 4.80e- 64 13      IL32    
## 67 1.03e- 66     2.00  0.296 0.013 2.08e- 62 13      KLRB1   
## 68 4.82e- 52     3.79  0.259 0.013 9.77e- 48 13      IGHG1   
## 69 3.98e- 44     3.37  0.259 0.015 8.06e- 40 13      IGHG4   
## 70 3.42e- 42     2.22  0.352 0.03  6.92e- 38 13      CCL5    
## 71 6.32e-208     1.85  0.978 0.036 1.28e-203 14      CPA3    
## 72 1.19e-176     1.28  0.822 0.03  2.41e-172 14      KIT     
## 73 3.11e-176     1.40  0.844 0.032 6.30e-172 14      MS4A2   
## 74 7.63e-172     1.75  0.889 0.036 1.54e-167 14      TPSAB1  
## 75 2.72e-158     2.05  0.978 0.051 5.50e-154 14      TPSB2</code></pre>
<p>For more details consult to <a href="http://satijalab01.nygenome.org/seurat/v3.0/de_vignette.html">DE vignette</a>.</p>
<p>According to the table above I generated a cluster id vector to assign cell type to each cluster. I have a group NA which I could not assign a cell type, which is probably a technical artifact from cell cycle regression since the top genes expressed in this population are cell cycle related. I also found 3 subpopulations of macrophages and 2 types of mast cells in thiS patient.</p>
<pre class="r"><code># Assigning Cell ids

new.cluster.ids &lt;- c(&quot;Macrophages(1)&quot;, &quot;Macrophages(2)&quot;, &quot;Macrophages(3)&quot;,
                     &quot;AT2(1) Cells&quot;, &quot;Cliated Cells&quot;, &quot;Basal Cells&quot;, &quot;Club Cells&quot;, 
                     &quot;Dendritic Cells&quot;, &quot;NA&quot;, &quot;AT2(2) Cells&quot;, &quot;Fibroblasts&quot;, 
                     &quot;Mast(1) Cells&quot;, &quot;Monocytes&quot;, &quot;T/NKT Cells&quot;, &quot;Mast(2) Cells&quot;)
names(new.cluster.ids) &lt;- levels(Fib)
Fib &lt;- RenameIdents(Fib, new.cluster.ids)
DimPlot(Fib, reduction = &quot;tsne&quot;, label = TRUE, label.size = 8, pt.size = 1.0) + NoLegend()</code></pre>
<p><img src="/post/2019-09-17-genomics-at-superresolution-advances-in-sequencing-technologies_files/figure-html/tsnewithcellids-1.png" width="1440" /></p>
<pre class="r"><code># Let&#39;s look at the expression of a few genes of interest accross different subtypes
VlnPlot(Fib, features = c(&quot;SPP1&quot;, &quot;CHIT1&quot;))</code></pre>
<p><img src="/post/2019-09-17-genomics-at-superresolution-advances-in-sequencing-technologies_files/figure-html/differentfeautures-1.png" width="672" /></p>
<pre class="r"><code>FeaturePlot(Fib, features = c(&quot;SPP1&quot;, &quot;APOE&quot;, &quot;SCGB3A2&quot;, &quot;C9orf24&quot;, &quot;S100A2&quot;, &quot;BPIFB1&quot;, &quot;S100B&quot;, &quot;COL3A1&quot;, &quot;FCN1&quot;))</code></pre>
<p><img src="/post/2019-09-17-genomics-at-superresolution-advances-in-sequencing-technologies_files/figure-html/differentfeautures-2.png" width="672" /></p>
</div>
<div id="what-cells-biotech-companies-are-targeting-in-lung-fibrosis" class="section level3">
<h3>What cells <code>Biotech</code> Companies are targeting in lung fibrosis?</h3>
<p>Novel therapies in pulmonary fibrosis are urgently needed. Let’s look at few promising drugs currently in clinical trials.</p>
<div id="fibrinogen" class="section level4">
<h4>Fibrinogen</h4>
<p>Fibrinoge’s lead compound pamrevlumab blocks CTGF. Let’s have a look which cells produce it.</p>
<pre class="r"><code>VlnPlot(Fib, features =c(&quot;CTGF&quot;))</code></pre>
<p><img src="/post/2019-09-17-genomics-at-superresolution-advances-in-sequencing-technologies_files/figure-html/Fibrinogen-1.png" width="672" /></p>
<pre class="r"><code>FeaturePlot(Fib, features = c(&quot;CTGF&quot;), pt.size = 1)</code></pre>
<p><img src="/post/2019-09-17-genomics-at-superresolution-advances-in-sequencing-technologies_files/figure-html/Fibrinogen-2.png" width="672" />
Main source for the <code>CTGF</code> production here seems the <code>Ciliated cells.</code></p>
</div>
<div id="galapagos" class="section level4">
<h4>Galapagos</h4>
<p>Galapagos GLPG1690 has entered Phase 3 clinical trials and it targets Autotaxin (ENPP2).</p>
<pre class="r"><code>VlnPlot(Fib, features =  c(&quot;ENPP2&quot;))</code></pre>
<p><img src="/post/2019-09-17-genomics-at-superresolution-advances-in-sequencing-technologies_files/figure-html/GLPG1690-1.png" width="672" /></p>
<pre class="r"><code>FeaturePlot(Fib, features =  c(&quot;ENPP2&quot;),  pt.size = 1)</code></pre>
<p><img src="/post/2019-09-17-genomics-at-superresolution-advances-in-sequencing-technologies_files/figure-html/GLPG1690-2.png" width="672" />
We see a low expression of <code>ENPP</code> mainly in Macrophages, Dendritic cells and Fibroblasts.</p>
<p>The other clinical candidate of Galapagos is GLPG1205 which targets GPR84. It has entered Phase 2 clinical trials.</p>
<pre class="r"><code>VlnPlot(Fib, features =  c(&quot;GPR84&quot;))</code></pre>
<p><img src="/post/2019-09-17-genomics-at-superresolution-advances-in-sequencing-technologies_files/figure-html/GLPG1205-1.png" width="672" /></p>
<pre class="r"><code>FeaturePlot(Fib, features =  c(&quot;GPR84&quot;), pt.size = 1)</code></pre>
<p><img src="/post/2019-09-17-genomics-at-superresolution-advances-in-sequencing-technologies_files/figure-html/GLPG1205-2.png" width="672" /></p>
<p><code>GPR84</code> has very low expression throughout the lung cells of this patient. The gene is mainly expressed in Type 1 Macrophages.</p>
</div>
<div id="bristol-myers-squibb-bms-986020" class="section level4">
<h4>Bristol-Myers-Squibb: <code>BMS-986020</code></h4>
<p>BMS-986020 is an anti-fibrotic drug being developed by Bristol-Myers Squibb, and is a lysophosphatidic acid (LPA) receptor antagonist.</p>
<p>Let’s look at the expression of a LPAR1.</p>
<pre class="r"><code>VlnPlot(Fib, features = c(&quot;LPAR1&quot;))</code></pre>
<p><img src="/post/2019-09-17-genomics-at-superresolution-advances-in-sequencing-technologies_files/figure-html/LPAR1-1.png" width="672" /></p>
<pre class="r"><code>FeaturePlot(Fib, features =c(&quot;LPAR1&quot;),  pt.size = 1)</code></pre>
<p><img src="/post/2019-09-17-genomics-at-superresolution-advances-in-sequencing-technologies_files/figure-html/LPAR1-2.png" width="672" />
It is expressed at low levels but consistently on many types of lung cells.</p>
</div>
<div id="roche-nintedanib" class="section level4">
<h4>Roche <code>nintedanib</code></h4>
<p>Nintedanib is one of the two new medicines approved for IPF patients. It targets multiple receptor tyrosine kinase receptors such as VEGF, FGF and PDGF.</p>
<pre class="r"><code>VlnPlot(Fib, features = c(&quot;VEGFA&quot;, &quot;PDGFA&quot;))</code></pre>
<p><img src="/post/2019-09-17-genomics-at-superresolution-advances-in-sequencing-technologies_files/figure-html/nintedanib-1.png" width="672" /></p>
<pre class="r"><code>FeaturePlot(Fib, features =  c(&quot;VEGFA&quot;, &quot;PDGFA&quot;), pt.size = 1)</code></pre>
<p><img src="/post/2019-09-17-genomics-at-superresolution-advances-in-sequencing-technologies_files/figure-html/nintedanib-2.png" width="672" /></p>
<p>VEGFA, PDGFA is highly expressed in multiple types of cells in the lung microenvironment.</p>
</div>
</div>
<div id="final-thoughts" class="section level3">
<h3>Final thoughts</h3>
<p>Seurat package allowed us to cluster different types of lung cells so that we can visualize their specific gene expression. We used those plots to see which cells express the drug targets in clinical trials or in use.<br />
Although changes associated with fibrosis are well defined the causal factors are not known. Reflecting this we saw that those <strong>drugs target many different components of the lung microenvironment.</strong> The conquest is on and we hope promising therapies arrive soon.</p>
</div>

    </div>

    


    

<div class="article-tags">
  
  <a class="badge badge-light" href="/tags/scrna-seq/">scRNA-seq</a>
  
  <a class="badge badge-light" href="/tags/seurat/">Seurat</a>
  
  <a class="badge badge-light" href="/tags/h5-file/">h5 file</a>
  
  <a class="badge badge-light" href="/tags/r/">R</a>
  
  <a class="badge badge-light" href="/tags/r-markdown/">R Markdown</a>
  
  <a class="badge badge-light" href="/tags/clustering/">Clustering</a>
  
  <a class="badge badge-light" href="/tags/pca/">PCA</a>
  
  <a class="badge badge-light" href="/tags/tsne/">tSNE</a>
  
  <a class="badge badge-light" href="/tags/ipf/">IPF</a>
  
</div>



    
      








  






  
  
  
    
  
  
  <div class="media author-card">
    
      
      <img class="portrait mr-3" src="/authors/admin/avatar_huc20273ad926c00177316852a0ac5eb17_12606_250x250_fill_q90_lanczos_center.jpg" alt="Avatar">
    

    <div class="media-body">
      <h5 class="card-title"><a href="/">Serdar Korur</a></h5>
      <h6 class="card-subtitle">Senior Scientist in Drug Discovery</h6>
      <p class="card-text">Serdar is a Data Scientist and PhD in Cell Biology</p>
      <ul class="network-icon" aria-hidden="true">
        
          
          
          
            
          
          
          
          
          
            
          
          <li>
            <a href="/#contact" >
              <i class="fas fa-envelope"></i>
            </a>
          </li>
        
          
          
          
            
          
          
          
          
          
            
          
          <li>
            <a href="https://www.linkedin.com/in/serdar-korur/" target="_blank" rel="noopener">
              <i class="fab fa-linkedin"></i>
            </a>
          </li>
        
          
          
          
            
          
          
          
          
          
            
          
          <li>
            <a href="https://github.com/korur/" target="_blank" rel="noopener">
              <i class="fab fa-github"></i>
            </a>
          </li>
        
      </ul>
    </div>
  </div>



      
      
      <div class="article-widget">
        <div class="hr-light"></div>
        <h3>Related</h3>
        <ul>
          
          <li><a href="/2019/09/15/deploy-your-blog-in-5-minutes/">Set up and deploy your blog in 5 minutes on Netlify with Hugo and blogdown (September 2019 Update)</a></li>
          
        </ul>
      </div>
      
    

    

    
<section id="comments">
  
    
<div id="disqus_thread"></div>
<script>
  let disqus_config = function () {
    
    
    
  };
  (function() {
    if (["localhost", "127.0.0.1"].indexOf(window.location.hostname) != -1) {
      document.getElementById('disqus_thread').innerHTML = 'Disqus comments not available by default when the website is previewed locally.';
      return;
    }
    var d = document, s = d.createElement('script'); s.async = true;
    s.src = 'https://' + "serdarkorur" + '.disqus.com/embed.js';
    s.setAttribute('data-timestamp', +new Date());
    (d.head || d.body).appendChild(s);
  })();
</script>
<noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
<a href="https://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>


  
</section>



  </div>
</article>

      

    
    
    
    <script src="/js/mathjax-config.js"></script>
    

    
    
    
      <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.imagesloaded/4.1.4/imagesloaded.pkgd.min.js" integrity="sha256-lqvxZrPLtfffUl2G/e7szqSvPBILGbwmsGE1MKlOi0Q=" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.isotope/3.0.6/isotope.pkgd.min.js" integrity="sha256-CBrpuqrMhXwcLLUd5tvQ4euBHCdh7wGlDfNz8vbu/iI=" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.2.5/jquery.fancybox.min.js" integrity="sha256-X5PoE3KU5l+JcX+w09p/wHl9AzK333C4hJ2I9S5mD4M=" crossorigin="anonymous"></script>

      

      
        
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.15.6/highlight.min.js" integrity="sha256-aYTdUrn6Ow1DDgh5JTc3aDGnnju48y/1c8s1dgkYPQ8=" crossorigin="anonymous"></script>
        
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.15.6/languages/r.min.js"></script>
        
      

      
      
      <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.4/MathJax.js?config=TeX-AMS_CHTML-full" integrity="sha256-GhM+5JHb6QUzOQPXSJLEWP7R73CbkisjzK5Eyij4U9w=" crossorigin="anonymous" async></script>
      
    

    
    
      <script src="https://cdnjs.cloudflare.com/ajax/libs/leaflet/1.2.0/leaflet.js" integrity="sha512-lInM/apFSqyy1o6s89K4iQUKg6ppXEgsVxT35HbzUupEVRh2Eu9Wdl4tHj7dZO0s1uvplcYGmt3498TtHq+log==" crossorigin="anonymous"></script>
    

    
    
    <script>hljs.initHighlightingOnLoad();</script>
    

    
    
    <script>
      const search_index_filename = "/index.json";
      const i18n = {
        'placeholder': "Search...",
        'results': "results found",
        'no_results': "No results found"
      };
      const content_type = {
        'post': "Posts",
        'project': "Projects",
        'publication' : "Publications",
        'talk' : "Talks"
        };
    </script>
    

    
    

    
    
    <script id="search-hit-fuse-template" type="text/x-template">
      <div class="search-hit" id="summary-{{key}}">
      <div class="search-hit-content">
        <div class="search-hit-name">
          <a href="{{relpermalink}}">{{title}}</a>
          <div class="article-metadata search-hit-type">{{type}}</div>
          <p class="search-hit-description">{{snippet}}</p>
        </div>
      </div>
      </div>
    </script>
    

    
    
    <script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/3.2.1/fuse.min.js" integrity="sha256-VzgmKYmhsGNNN4Ph1kMW+BjoYJM2jV5i4IlFoeZA9XI=" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script>
    

    
    

    
    
    <script id="dsq-count-scr" src="https://serdarkorur.disqus.com/count.js" async></script>
    

    
    
    
    
    
    
    
    
    
      
    
    
    
    
    <script src="/js/academic.min.130521ecfc6f534c52c158217bbff718.js"></script>

    






  
  <div class="container">
    <footer class="site-footer">
  
  <p class="powered-by">
    
      <a href="/privacy/">Privacy Policy</a>
    
    
       &middot; 
      <a href="/terms/">Terms</a>
        
  </p>
  

  <p class="powered-by">
    

    Powered by the
    <a href="https://sourcethemes.com/academic/" target="_blank" rel="noopener">Academic theme</a> for
    <a href="https://gohugo.io" target="_blank" rel="noopener">Hugo</a>.

    
    <span class="float-right" aria-hidden="true">
      <a href="#" id="back_to_top">
        <span class="button_icon">
          <i class="fas fa-chevron-up fa-2x"></i>
        </span>
      </a>
    </span>
    
  </p>
</footer>

  </div>
  

  
<div id="modal" class="modal fade" role="dialog">
  <div class="modal-dialog">
    <div class="modal-content">
      <div class="modal-header">
        <h5 class="modal-title">Cite</h5>
        <button type="button" class="close" data-dismiss="modal" aria-label="Close">
          <span aria-hidden="true">&times;</span>
        </button>
      </div>
      <div class="modal-body">
        <pre><code class="tex hljs"></code></pre>
      </div>
      <div class="modal-footer">
        <a class="btn btn-outline-primary my-1 js-copy-cite" href="#" target="_blank">
          <i class="fas fa-copy"></i> Copy
        </a>
        <a class="btn btn-outline-primary my-1 js-download-cite" href="#" target="_blank">
          <i class="fas fa-download"></i> Download
        </a>
        <div id="modal-error"></div>
      </div>
    </div>
  </div>
</div>

</body>
</html>
